consensus/src/execution_ctx.rs

```diff
@@ -322,7 +336,11 @@ impl<'a, T: Operations + 'static> ExecutionCtx<'a, T> {
             Err(ConsensusError::FutureEvent) => {
                 trace!("future msg {:?}", msg);
 
-                self.outbound.try_send(msg.clone());
+                // Re-propagate messages from future iterations of the current
+                // round
+                if msg.header.round == self.round_update.round {
+                    self.outbound.try_send(msg.clone());
+                }
 
```

Comment:
It is not discarded. It's put in a queue and processed at the right round (where it will be re-propagated).

Messages that are more than 10 rounds in the future are indeed discarded, but that's done before reaching this point